src/Mvc/Mvc.Analyzers/src/Startup/MiddlewareOrderingAnalzyer.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.AspNetCore.Analyzers
+{
+    internal class MiddlewareOrderingAnalyzer
+    {
+        // This should probably be a multi-map, but oh-well.
+        private readonly static ImmutableDictionary<string, string> MiddlewareHappensAfterMap = ImmutableDictionary.CreateRange<string, string>(new[]
+        {
+            new KeyValuePair<string, string>("UseAuthorization", "UseAuthentication"),
+        });
```

Comment:
This is the kind of info we want to get from the runtime so the analyzer can support all versions. For now I'm just pointing that out.